platform/migration/platform-migratable-util/src/main/java/org/codice/ddf/migration/util/MigratableUtil.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration.util;
+
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationWarning;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class for services that implement {@link org.codice.ddf.migration.Migratable}.
+ * <p/>
+ * <p>
+ * <b>This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library.</b>
+ * </p>
+ */
+public class MigratableUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigratableUtil.class);
+
+    private static final String UNREAL_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that could not coerced into a real path; "
```

Comment:
Ya, this and most of the previous comments have been addressed in Eric's PR, these commits are a bit outdated. My changes are just within the security-migratable module which should be in the last commit only. I mentioned it in the PR's description, but I'm gonna go ahead and just take out those commits for this PR's branch to avoid more confusion.